src/stdlib/parse_auditd.rs

```diff
@@ -0,0 +1,435 @@
+
+        avc_denied {
+            args: func_args![value: r#"type=AVC msg=audit(1724423274.618:6439): avc:  denied  { setuid setuid2 setuid3 } for  pid=1240242 comm="vrl" capability=7  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=1"#],
+            want: Ok(btreemap! {
+                "type" => "AVC",
+                "timestamp" => DateTime::from_timestamp_millis(1_724_423_274_618),
+                "sequence" => 6439,
+                "body" => btreemap! {
+                    "denied" => vec!["setuid", "setuid2", "setuid3"],
+                    "pid" => 1_240_242,
+                    "comm" => "vrl",
+                    "capability" => 7,
+                    "scontext" => "system_u:system_r:auditd_t:s0",
+                    "tcontext" => "system_u:system_r:auditd_t:s0",
+                    "tclass" => "capability",
+                    "permissive" => 1
+                }
+            }),
+            tdef: type_def(),
+        }
+
+        avc_granted {
+            args: func_args![value: r#"type=AVC msg=audit(1724423274.618:6439): avc:  granted  { setsecparam setsecparam2 setsecparam3} for  pid=1240242 comm="vrl" scontext=system_u:system_r:vrl_t:s0 tcontext=system_u:object_r:security_t:s0 tclass=security"#],
+            want: Ok(btreemap! {
+                "type" => "AVC",
+                "timestamp" => DateTime::from_timestamp_millis(1_724_423_274_618),
+                "sequence" => 6439,
+                "body" => btreemap! {
+                    "granted" => vec!["setsecparam","setsecparam2","setsecparam3"],
+                    "pid" => 1_240_242,
+                    "comm" => "vrl",
+                    "scontext" => "system_u:system_r:vrl_t:s0",
+                    "tcontext" => "system_u:object_r:security_t:s0",
+                    "tclass" => "security"
+                }
+            }),
+            tdef: type_def(),
+        }
+
+        user_acct {
+            args: func_args![value: format!(r#"type=USER_ACCT msg=audit(1724423274.618:6439): pid=1240242 uid=1000 auid=1000 ses=2 msg='op=PAM:accounting grantors=pam_unix,pam_permit,pam_time acct="vrl" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/1 res=success'{}UID="vrl" AUID="vrl""#, ENRICHMENT_SEPARATOR)],
+            want: Ok(btreemap! {
+                "type" => "USER_ACCT",
+                "timestamp" => DateTime::from_timestamp_millis(1_724_423_274_618),
+                "sequence" => 6439,
+                "body" => btreemap! {
+                    "pid" => 1_240_242,
+                    "uid" => 1_000,
+                    "auid" => 1_000,
+                    "ses" => 2,
+                    "msg"=> btreemap! {
+                        "op" => "PAM:accounting",
+                        "grantors" => "pam_unix,pam_permit,pam_time",
+                        "acct" => "vrl",
+                        "exe" => "/usr/bin/sudo",
+                        "hostname" => Value::Null,
```

Comment:
Fields that are missing in the audit log (for example, `hostname="?"`) are parsed into `Null` values.

They are shown like this in vector:
```json
{
    "body": {
        "auid": 4294967295,
        "msg": {
            "addr": null,
            "comm": "systemd",
            "exe": "/usr/lib/systemd/systemd",
            "hostname": null,
            "res": "success",
            "terminal": null,
            "unit": "wpa_supplicant"
        },
        "pid": 1,
        "ses": 4294967295,
        "uid": 0
    },
    "enrichment": {
        "AUID": "unset",
        "UID": "root"
    },
    "sequence": 165,
    "timestamp": "2024-08-30T18:14:22.240Z",
    "type": "SERVICE_STOP"
}
```

Should we filter out that fields, or leave the entry with a null value, so people know that it is present but with no value?